tests/test_basic.py

```diff
@@ -40,41 +44,98 @@ async def test_basic_path(joint_fixture: JointFixture):  # noqa: F811
     assert plaintext_content.strip() == expected_plaintext
 
-    html_body = msg.get_body(preferencelist=("html"))
-    assert html_body is not None
+    if msg.is_multipart():
+        html_body = msg.get_body(preferencelist="html")
+        assert html_body is not None
+
+        html_content = html_body.get_content()  # type: ignore[attr-defined]
+        assert html_content is not None
+
+        expected_html = '<!DOCTYPE html><html><head></head><body style="color: #00393f;padding: 12px;"><h2>Dear Yolanda Martinez,</h2><p>Where are you, where are you, Yolanda?</p><p>Warm regards,</p><h3>The GHGA Team</h3></body></html>'
+        assert html_content.strip() == expected_html
+
+
+@pytest.mark.parametrize(
+    "notification_details",
+    [
+        {
+            "recipient_email": "test@example.com",
+            "email_cc": ["test2@test.com", "test3@test.com"],
+            "email_bcc": ["test4@test.com", "test5@test.com"],
+            "subject": "Test123",
+            "recipient_name": "Yolanda Martinez",
+            "plaintext_body": "Where are you, where are you, Yolanda?",
+        }
+    ],
+)
+@pytest.mark.asyncio
+async def test_transmission(notification_details):
+    """Test that the email that the test server gets is what we expect"""
+    config = get_config()
+    notification = make_notification(notification_details)
+    port_to_use = get_free_port()
+
+    dummy_smtp_client = DummySmtpClient()
+    smtp_client = SmtpClient(config=config, debugging=True)
+    smtp_client.set_port(port_to_use)
+
+    server = DummyServer(config=config)
+    server.set_port(port_to_use)
+
+    notifier = Notifier(config=config, smtp_client=dummy_smtp_client)
+
+    # send the notification so it gets intercepted by the dummy client
+    await notifier.send_notification(notification=notification)
 
-    html_content = html_body.get_content()  # type: ignore[attr-defined]
-    assert html_content is not None
+    # tell the smtp client to send the message and compare that with what is received
+    async with server.expect_email(expected_email=dummy_smtp_client.expected_email):
+        smtp_client.send_email_message(dummy_smtp_client.expected_email)
+        asyncio.get_running_loop().stop()
 
-    expected_html = '<!DOCTYPE html><html><head></head><body style="color: #00393f;padding: 12px;"><h2>Dear Yolanda Martinez,</h2><p>Where are you, where are you, Yolanda?</p><p>Warm regards,</p><h3>The GHGA Team</h3></body></html>'
-    assert html_content.strip() == expected_html
+    # change login credentials so authentication fails
+    server.set_credentials(login="bob@bobswebsite.com", password="notCorrect")
+    with pytest.raises(SmtpClient.FailedLoginError):
+        async with server.expect_email(expected_email=dummy_smtp_client.expected_email):
+            smtp_client.send_email_message(dummy_smtp_client.expected_email)
+            asyncio.get_running_loop().stop()
```

Comment:
Usually we try to separate happy case from error case testing. Better to put this into a separate test.